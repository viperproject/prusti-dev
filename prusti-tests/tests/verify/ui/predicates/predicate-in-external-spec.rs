use prusti_contracts::*;

trait MyTrait {
    fn foo() -> bool;
}

#[extern_spec]
trait MyTrait {
    predicate!{
        fn foo() -> bool;
    }
}

struct MyStruct;
impl MyStruct {
    fn bar() -> bool {
        true
    }
}

#[extern_spec]
impl MyStruct {
    predicate!{
        fn bar() -> bool;
    }
}

fn main() {}
