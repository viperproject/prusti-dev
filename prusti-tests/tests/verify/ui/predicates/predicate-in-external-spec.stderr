error: Can not declare abstract predicate in external spec
  --> $DIR/predicate-in-external-spec.rs:9:5
   |
9  | /     predicate!{
10 | |         fn foo() -> bool;
11 | |     }
   | |_____^

error: Can not declare abstract predicate in external spec
  --> $DIR/predicate-in-external-spec.rs:23:5
   |
23 | /     predicate!{
24 | |         fn bar() -> bool;
25 | |     }
   | |_____^

error: aborting due to 2 previous errors

